### PR TITLE
Fix EventScheduler Poisson.

### PR DESCRIPTION
**Pull request by jessetilro**
Updated the EventScheduler update algorithm to an implementation of pseudocode for simulating a
poisson process (reference will be included in the pull-request). Also increased precision, using 64 bit floating point numbers rather than 32 bit. Finally, this implementation allows for multiple events to occur simultaneously (with a really small probability though).

After grabbing my book on probability and statistics I noticed that the EventScheduler wasn't functioning correct yet. Therefore I implemented the first of these algorithms:
[https://en.wikipedia.org/wiki/Poisson_distribution#Generating_Poisson-distributed_random_variables](https://en.wikipedia.org/wiki/Poisson_distribution#Generating_Poisson-distributed_random_variables)

### CODE REVIEW

File: src/main/java/pixelperfect/event/EventScheduler.java [hunk 1]
```diff
@@ -54,15 +54,27 @@ private void publish(Event event) {
    * models a poisson process, with the specified intensity as lambda parameter.
    * 
    * @param tpf
-   *          The time per frame in seconds, i.e. how many times per second this draw is performed
-   *          and thus the time interval parameter for the poisson process.
+   *          The time per frame in seconds, i.e. the time interval parameter for the poisson
+   *          process.
    */
   public void update(float tpf) {
+    // The random generator should be more efficient and less biased than the Math.random function.
```
Comment (dalderliesten) [inline src/main/java/pixelperfect/event/EventScheduler.java]: I know it's "mierenneukerij," but gramatically it should be "This random generator is more efficient and...".

**Comment by dalderliesten** [comment]
Looks good to me, all up to Wouter now.

**Comment by FlorisXll** [comment]
This probably needs to be finetuned in the very end, because now 6 events can actually at the same time, if you are a believer. We can just threshold it easily.

**Comment by wouterz** [comment]
Looks good, only suggestion would be having the intensity as a variable that can be set for multiple difficulties. But this can also easily be done later on

**Pull request merged**